Cabal/Distribution/Simple/Build/Macros.hs

```diff
@@ -73,7 +76,8 @@ generate :: PackageDescription -> LocalBuildInfo -> ComponentLocalBuildInfo -> S
 generate pkg_descr lbi clbi =
   "/* DO NOT EDIT: This file is automatically generated by Cabal */\n\n" ++
   generatePackageVersionMacros
-    (package pkg_descr : map snd (componentPackageDeps clbi)) ++
+    (computeCompatPackageId (package pkg_descr) CLibName
```

Comment:
Yeah, it's a no-op. I like to convert in this way this as a warning sign that the code might not be very component-aware.